src/snowflake/snowpark/session.py

```diff
@@ -4071,10 +4078,16 @@ def telemetry_enabled(self, value):
         # Set both in-band and out-of-band telemetry to True/False
         if value:
             self._conn._conn.telemetry_enabled = True
-            self._conn._telemetry_client.telemetry._enabled = True
+            self._conn._telemetry_client._enabled = True
+            if is_in_stored_procedure() and not self._stored_proc_telemetry_enabled:
+                _logger.debug(
+                    "Client side parameter ENABLE_SNOWPARK_FIRST_PARTY_TELEMETRY is set to False, telemetry could not be enabled"
+                )
+                self._conn._telemetry_client._enabled = False
+
         else:
             self._conn._conn.telemetry_enabled = False
-            self._conn._telemetry_client.telemetry._enabled = False
+            self._conn._telemetry_client._enabled = False
```

Comment:
Isn't having `self._conn._telemetry_client._enabled` sufficient here. We can get rid of `self._conn._conn.telemetry_enabled`